// mongodb-stitch-browser-core/dist/esm/core/internal/StitchAppClientImpl.js
import { CoreStitchAppClient, StitchAppClientInfo, StitchRequestClient } from "mongodb-stitch-core-sdk";
import StitchServiceImpl from "../../services/internal/StitchServiceImpl";
import StitchAuthImpl from "../auth/internal/StitchAuthImpl";
import StitchBrowserAppRoutes from "../auth/internal/StitchBrowserAppRoutes";
var StitchAppClientImpl = (function () {
    function StitchAppClientImpl(clientAppId, config) {
        this.info = new StitchAppClientInfo(clientAppId, config.dataDirectory, config.localAppName, config.localAppVersion);
        this.routes = new StitchBrowserAppRoutes(this.info.clientAppId, config.baseUrl);
        var requestClient = new StitchRequestClient(config.baseUrl, config.transport);
        this.auth = new StitchAuthImpl(requestClient, this.routes.authRoutes, config.storage, this.info);
        this.coreClient = new CoreStitchAppClient(this.auth, this.routes);
    }
    StitchAppClientImpl.prototype.getServiceClient = function (factory, serviceName) {
        if (isServiceClientFactory(factory)) {
            return factory.getClient(new StitchServiceImpl(this.auth, this.routes.serviceRoutes, ""), this.info);
        }
        else {
            return factory.getNamedClient(new StitchServiceImpl(this.auth, this.routes.serviceRoutes, serviceName), this.info);
        }
    };
    StitchAppClientImpl.prototype.callFunction = function (name, args) {
        return this.coreClient.callFunctionInternal(name, args);
    };
    return StitchAppClientImpl;
}());
export default StitchAppClientImpl;
function isServiceClientFactory(factory) {
    return factory.getClient !== undefined;
}
